web/websearch.py

```diff
@@ -918,8 +918,226 @@ def nl2br(value):
+        result['debug']['queries'] = []
+
+    #SphinxSearch connection details
+    host = '127.0.0.1'
+    port = 9306
+    try:
+        conn = MySQLdb.connect(host=host, port=port, user='root')
+        curs = conn.cursor()
+    except Exception as ex:
+        status = False
+        result['message'] = str(ex)
+        result['status'] = status
+        return result, 0
+
+    # We attempt to find rows using a small bounding box to
+    # limit the impact of the distance calculation.
+    # If no rows are found with the current bounding box
+    # we double it and try again, until a result is returned.
+
+    delta = 0.0004
+    count = 0
+
+    while count==0:
+        delta *= 2
+        lon_min = lon - delta
+        lon_max = lon + delta
+        lat_min = lat - delta
+        lat_max = lat + delta
+
+        # Bound the latitude
+        lat_min = max(min(lat_min,90.0),-90.0)
+        lat_max = max(min(lat_max,90.0),-90.0)
+
+        #we use the built-in GEODIST function to calculate distance
+        select = "SELECT *, GEODIST("+str(lat)+","+str(lon)+",lat,lon,{in=degrees, out=meters}) as distance \
+                     FROM ind_name_exact "
+
+        # Handle 180 meridian
+        if (lon_min < -180.0):
+            """
+            SphinxQL does not support the OR operator
+            or the NOT BETWEEN syntax so the only
+            viable approach is to use 2 queries
+
+            However we still set the sql variable and use this
+            for debug output purposes
+            """
+
+            where = "WHERE ((lon BETWEEN {} AND 180.0) \
+                       OR  (lon BETWEEN -180.0 AND {})) ".format(360.0+lon_min, lon_max)
+
+            where1 = "WHERE lon BETWEEN {} AND 180.0 ".format(360.0+lon_min)
+
+            where2 = "WHERE lon BETWEEN -180.0 AND {} ".format(lon_max)
+
+        elif (lon_max > 180.0):
+            """
+            SphinxQL does not support the OR operator
+            or the NOT BETWEEN syntax so the only
+            viable approach is to use 2 queries
+
+            However we still set the sql variable and use this
+            for debug output purposes
+            """
+
+            where = "WHERE ((lon BETWEEN {} AND 180.0) \
+                       OR  (lon BETWEEN -180.0 AND {})) ".format(lon_min, -360.0+lon_max)
+
+            where1 = "WHERE lon BETWEEN {} AND 180.0 ".format(lon_min)
+
+            where2 = "WHERE lon BETWEEN -180.0 AND {} ".format(-360.0+lon_max)
+
+        else:
+            #normal case
+            where = "WHERE lon BETWEEN {} AND {} ".format(lon_min, lon_max)
+            where1 = ""
+            where2 = ""
 
+        #latitude condition is fixed
+        wherelat = "AND lat BETWEEN {} AND {} AND class='place'".format(lat_min, lat_max)
 
+        #limit the result set to the single closest match
+        limit = " ORDER BY distance ASC LIMIT 1"
+
+        #form the final query/queries
+        sql = select + where + wherelat + limit
```

Comment:
I am not quite happy with this code, it's very repeating and using a lot of variables.
It could be rewritten to more "generic way" and easy to read.